### ISSUE TITLE
Reverting to vanilla Mastodon

### ISSUE THREAD

**Issue by brortao** [glitch-soc/mastodon#1203]
Is it possible to migrate from glitch-soc to vanilla Mastodon? If so, perhaps the docs should have instructions on this (e.g. if you need to run any SQL queries, etc.)

**Comment by ThibG** [comment]
It should be possible, but there is no procedure to do that. If you do so, you should expect the following issues (from the top of my head):
- Any toot that is not plaintext will now display its source
- Any local-only toot will no longer be local-only
- Database will contain orphaned columns and tables, which should not be a huge issue though

**Comment by vibhi19** [comment]
My friend Just switched and There are some serious issue with webUI. The UI is very laggy and slow, he even did a complete fresh install of Vanilla but same issues.

**Comment by ThibG** [comment]
> My friend Just switched and There are some serious issue with webUI. The UI is very laggy and slow, he even did a complete fresh install of Vanilla but same issues.

I am confused. If a fresh install of Vanilla has the same issues, how is that related to glitch-soc?

**Comment by vibhi19** [comment]
By Fresh install of vanilla i mean he used old database of Glitch. it was basically a migration to new server. now he moved to glitch again and everything is good again

**Comment by brortao** [comment]
I did this with no issues. The steps I took were:

* Update glitch to latest master
* Back up the database
* Clone vanilla Mastodon (also from latest master)
* From the glitch directory, roll back any migrations that aren't in vanilla Mastodon
  * Use `diff -rq mastodon/db/migrate glitch/db/migrate` to find these
  * Then use `RAILS_ENV=production bundle exec db:migrate:down VERSION=20190101000000`, for each of the migrations

That's it -- then just install normally (`bundle install`, precompile, etc.) from the vanilla Mastodon directory.

I'm not seeing any of the slowdown issues that @vibhi19 mentions.
